runtime/src/iree/hal/drivers/cuda/registration/driver_module.c

```diff
@@ -39,6 +39,34 @@ static iree_status_t iree_hal_cuda_driver_factory_enumerate(
   return iree_ok_status();
 }
 
+#if IREE_HAL_DRIVER_CUDA_NCCL
```

Comment:
you can remove the #ifdef here - the options are always present and it's fine to parse them - the less conditionally compiled code we have the better and we really just want to guard against needing the nccl headers or large additional code inclusions